plugins/modules/provision_workflow_manager.py

```diff
@@ -802,23 +836,39 @@ def get_wireless_params(self):
             }
         ]
 
-        if self.compare_dnac_versions(self.get_ccc_version(), "2.3.5.3") <= 0:
-            self.log("Checking for managed AP locations in Catalyst Center version <= 2.3.5.3", "DEBUG")
-            ap_locations = wireless_params[0].get("managedAPLocations")
-
-            if not (ap_locations and isinstance(ap_locations, list)):
-                self.log("Validating AP locations: {0}".format(ap_locations), "DEBUG")
-                msg = "Missing Managed AP Locations: Please specify the intended location(s) for the wireless device \
-                    within the site hierarchy."
-                self.log(msg, "CRITICAL")
-                self.module.fail_json(msg=msg, response=[])
-
-            for ap_loc in self.validated_config.get("managed_ap_locations"):
-                site_type = self.get_site_type(site_name_hierarchy=ap_loc)
-                self.log("Checking site type for AP location '{0}', resolved type: '{1}'".format(ap_loc, site_type), "DEBUG")
-                if site_type != "floor":
-                    self.log("Managed AP Location must be a floor", "CRITICAL")
-                    self.module.fail_json(msg="Managed AP Location must be a floor", response=[])
+        ap_locations = wireless_params[0].get("managedAPLocations")
+        if not (ap_locations and isinstance(ap_locations, list)):
+            self.log("Validating AP locations: {0}".format(ap_locations), "DEBUG")
+            msg = "Missing Managed AP Locations: Please specify the intended location(s) for the wireless device \
+                within the site hierarchy."
+            self.log(msg, "CRITICAL")
+            self.module.fail_json(msg=msg, response=[])
+
+        self.floor_names = []
+        for ap_loc in self.validated_config.get("managed_ap_locations"):
+            site_type = self.get_site_type(site_name_hierarchy=ap_loc)
+            self.log("Checking site type for AP location '{0}', resolved type: '{1}'".format(ap_loc, site_type), "DEBUG")
+
+            if site_type not in ["floor", "building"]:
```

Comment:
The site_type could be either floor or building or otherwise.. Can we simplify it?

```
    if site_type == "floor":
        self.log("Adding '{0}' to floor names list".format(ap_loc), "DEBUG")
        self.floor_names.append(ap_loc)
    elif site_type == "building":
        self.log("Building site type detected for '{0}'. Retrieving floor details.".format(ap_loc), "DEBUG")
        building_name = ap_loc + ".*"
        floors = self.get_site(building_name)

        for item in floors['response']:
            if item.get('type') == 'floor':
                self.log("Floor found: '{0}' for building '{1}'".format(item['nameHierarchy'], ap_loc), "DEBUG")
                self.floor_names.append(item['nameHierarchy'])
            elif 'additionalInfo' in item:
                for additional_info in item['additionalInfo']:
                    if 'attributes' in additional_info and additional_info['attributes'].get('type') == 'floor':
                        self.log("Floor found in additionalInfo: '{0}' for building '{1}'".format(item['siteNameHierarchy'], ap_loc), "DEBUG")
                        self.floor_names.append(item['siteNameHierarchy'])
    else:
        self.log("Invalid site type detected for '{0}'. Managed AP Location must be building or floor.".format(ap_loc), "CRITICAL")
        self.module.fail_json(msg="Managed AP Location must be building or floor", response=[])

self.log("Final list of floor names: {0}".format(self.floor_names), "DEBUG")
```

Reply:
fixed the code